packages/cdktf/test/functions.test.ts

```diff
@@ -0,0 +1,357 @@
+import { Testing, TerraformStack, TerraformOutput, Fn } from "../lib";
+import { TerraformVariable } from "../lib/terraform-variable";
+
+test("static values", () => {
+  const app = Testing.app();
+  const stack = new TerraformStack(app, "test");
+
+  new TerraformOutput(stack, "test-output", {
+    value: Fn.abs(-42),
+  });
+  expect(Testing.synth(stack)).toMatchInlineSnapshot(`
+    "{
+      \\"//\\": {
+        \\"metadata\\": {
+          \\"version\\": \\"stubbed\\",
+          \\"stackName\\": \\"test\\",
+          \\"backend\\": \\"local\\"
+        }
+      },
+      \\"output\\": {
+        \\"test-output\\": {
+          \\"value\\": \\"\${abs(-42)}\\"
+        }
+      }
+    }"
+  `);
+});
+
+test("dynamic values", () => {
+  const app = Testing.app();
+  const stack = new TerraformStack(app, "test");
+
+  const variable = new TerraformVariable(stack, "test-var", {
+    type: "number",
+  });
+
+  new TerraformOutput(stack, "test-output", {
+    value: Fn.abs(variable.value),
+  });
+  expect(Testing.synth(stack)).toMatchInlineSnapshot(`
+    "{
+      \\"//\\": {
+        \\"metadata\\": {
+          \\"version\\": \\"stubbed\\",
+          \\"stackName\\": \\"test\\",
+          \\"backend\\": \\"local\\"
+        }
+      },
+      \\"variable\\": {
+        \\"test-var\\": {
+          \\"type\\": \\"number\\"
+        }
+      },
+      \\"output\\": {
+        \\"test-output\\": {
+          \\"value\\": \\"\${abs(var.test-var)}\\"
+        }
+      }
+    }"
+  `);
+});
+
+test("spreaded mixed values", () => {
+  const app = Testing.app();
+  const stack = new TerraformStack(app, "test");
+
+  const variable = new TerraformVariable(stack, "test-var", {
+    type: "number",
+  });
+
+  new TerraformOutput(stack, "test-output", {
+    value: Fn.max([10, variable.value, 200]),
```

Comment:
Passing an array is a bit clunky.

Reply:
Yeah, I'm also not a huge fan, but JSII seemed to have problems with spread parameters 🤔

Reply:
It looks as though jsii is intended to support variadic arguments, but there may be some issues.